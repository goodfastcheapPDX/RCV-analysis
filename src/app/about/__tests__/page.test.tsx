import { render } from "@testing-library/react";
import { describe, expect, it } from "vitest";
import AboutPage from "../page";

describe("About Page", () => {
  it("should match snapshot", () => {
    const { container } = render(<AboutPage />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
