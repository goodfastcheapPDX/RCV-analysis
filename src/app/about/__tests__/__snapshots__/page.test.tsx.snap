// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`About Page > should match snapshot 1`] = `
<div
  class="space-y-6"
>
  <div>
    <h1
      class="text-3xl font-bold"
    >
      About
    </h1>
    <p
      class="text-muted-foreground mt-2"
    >
      Comprehensive analysis platform for ranked-choice voting elections
    </p>
  </div>
  <div
    class="bg-card text-card-foreground flex flex-col gap-6 rounded-xl border py-6 shadow-sm"
    data-slot="card"
  >
    <div
      class="@container/card-header grid auto-rows-min grid-rows-[auto_auto] items-start gap-1.5 px-6 has-data-[slot=card-action]:grid-cols-[1fr_auto] [.border-b]:pb-6"
      data-slot="card-header"
    >
      <div
        class="leading-none font-semibold"
        data-slot="card-title"
      >
        Ranked Elections Analyzer
      </div>
      <div
        class="-foreground text-sm"
        data-slot="card-description"
      >
        Advanced tools for understanding ranked-choice voting patterns and outcomes
      </div>
    </div>
    <div
      class="px-6 space-y-4"
      data-slot="card-content"
    >
      <p>
        This platform provides comprehensive analysis tools for ranked-choice voting elections, featuring Single Transferable Vote (STV) tabulation, coalition analysis, and interactive visualizations.
      </p>
      <div
        class="grid gap-4 sm:grid-cols-2"
      >
        <div>
          <h3
            class="font-semibold mb-2"
          >
            Key Features
          </h3>
          <ul
            class="space-y-1 text-sm text-muted-foreground"
          >
            <li>
              • STV tabulation engine
            </li>
            <li>
              • Coalition analysis
            </li>
            <li>
              • Interactive visualizations
            </li>
            <li>
              • Cast vote record processing
            </li>
          </ul>
        </div>
        <div>
          <h3
            class="font-semibold mb-2"
          >
            Technology Stack
          </h3>
          <div
            class="flex flex-wrap gap-2"
          >
            <span
              class="inline-flex items-center justify-center rounded-md border px-2 py-0.5 text-xs font-medium w-fit whitespace-nowrap shrink-0 [&>svg]:size-3 gap-1 [&>svg]:pointer-events-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive transition-[color,box-shadow] overflow-hidden border-transparent bg-secondary text-secondary-foreground [a&]:hover:bg-secondary/90"
              data-slot="badge"
            >
              Next.js
            </span>
            <span
              class="inline-flex items-center justify-center rounded-md border px-2 py-0.5 text-xs font-medium w-fit whitespace-nowrap shrink-0 [&>svg]:size-3 gap-1 [&>svg]:pointer-events-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive transition-[color,box-shadow] overflow-hidden border-transparent bg-secondary text-secondary-foreground [a&]:hover:bg-secondary/90"
              data-slot="badge"
            >
              TypeScript
            </span>
            <span
              class="inline-flex items-center justify-center rounded-md border px-2 py-0.5 text-xs font-medium w-fit whitespace-nowrap shrink-0 [&>svg]:size-3 gap-1 [&>svg]:pointer-events-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive transition-[color,box-shadow] overflow-hidden border-transparent bg-secondary text-secondary-foreground [a&]:hover:bg-secondary/90"
              data-slot="badge"
            >
              Tailwind CSS
            </span>
            <span
              class="inline-flex items-center justify-center rounded-md border px-2 py-0.5 text-xs font-medium w-fit whitespace-nowrap shrink-0 [&>svg]:size-3 gap-1 [&>svg]:pointer-events-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive transition-[color,box-shadow] overflow-hidden border-transparent bg-secondary text-secondary-foreground [a&]:hover:bg-secondary/90"
              data-slot="badge"
            >
              DuckDB
            </span>
            <span
              class="inline-flex items-center justify-center rounded-md border px-2 py-0.5 text-xs font-medium w-fit whitespace-nowrap shrink-0 [&>svg]:size-3 gap-1 [&>svg]:pointer-events-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive transition-[color,box-shadow] overflow-hidden border-transparent bg-secondary text-secondary-foreground [a&]:hover:bg-secondary/90"
              data-slot="badge"
            >
              Zod
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="grid gap-6 md:grid-cols-2"
  >
    <div
      class="bg-card text-card-foreground flex flex-col gap-6 rounded-xl border py-6 shadow-sm"
      data-slot="card"
    >
      <div
        class="@container/card-header grid auto-rows-min grid-rows-[auto_auto] items-start gap-1.5 px-6 has-data-[slot=card-action]:grid-cols-[1fr_auto] [.border-b]:pb-6"
        data-slot="card-header"
      >
        <div
          class="leading-none font-semibold"
          data-slot="card-title"
        >
          Data Sources
        </div>
        <div
          class="-foreground text-sm"
          data-slot="card-description"
        >
          Election data and formats
        </div>
      </div>
      <div
        class="px-6"
        data-slot="card-content"
      >
        <p
          class="text-muted-foreground text-sm"
        >
          Supports Cast Vote Record (CVR) data in CSV format for comprehensive ballot-level analysis of ranked-choice elections.
        </p>
      </div>
    </div>
    <div
      class="bg-card text-card-foreground flex flex-col gap-6 rounded-xl border py-6 shadow-sm"
      data-slot="card"
    >
      <div
        class="@container/card-header grid auto-rows-min grid-rows-[auto_auto] items-start gap-1.5 px-6 has-data-[slot=card-action]:grid-cols-[1fr_auto] [.border-b]:pb-6"
        data-slot="card-header"
      >
        <div
          class="leading-none font-semibold"
          data-slot="card-title"
        >
          Use Cases
        </div>
        <div
          class="-foreground text-sm"
          data-slot="card-description"
        >
          Who benefits from this platform
        </div>
      </div>
      <div
        class="px-6"
        data-slot="card-content"
      >
        <ul
          class="space-y-1 text-sm text-muted-foreground"
        >
          <li>
            • Election administrators
          </li>
          <li>
            • Political researchers
          </li>
          <li>
            • Campaign strategists
          </li>
          <li>
            • Civic organizations
          </li>
          <li>
            • Curious citizens
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
